keep/api/arq_worker.py

```diff
@@ -79,6 +82,22 @@ def get_arq_worker(queue_name: str) -> Worker:
     )
     return worker
 
+async def safe_run_worker(worker: Worker):
+    try:
+        while True:
+            try:
+                await worker.async_run()
+            except asyncio.CancelledError:  # pragma: no cover
+                # happens on shutdown, fine
+                pass
+            except redis.exceptions.ConnectionError:
+                logger.exception("Failed to connect to Redis... Retry in 3 seconds")
+                await asyncio.sleep(3)
+                continue
+            break
+    finally:
+        await worker.close()
```

Comment:
maybe add info logs on shuwdown and finally?

Reply:
We have lower-level logs from the lib